platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -651,6 +690,12 @@ - (id)mgl_jsonExpressionObject {
                 return @[@"to-string", self.operand.mgl_jsonExpressionObject];
             } else if ([function isEqualToString:@"noindex:"]) {
                 return self.arguments.firstObject.mgl_jsonExpressionObject;
+            } else if ([function isEqualToString:@"mgl_has:"]) {
+                NSMutableArray *expressionObject = [NSMutableArray arrayWithObjects:@"has", self.arguments[0].mgl_jsonExpressionObject, nil];
+                if (self.operand.expressionType == NSConstantValueExpressionType) {
```

Comment:
`FUNCTION($featureProperties, 'mgl_has:', 'property')` should be translated as `["has", ["properties"], "property"]`, so we need to check here that the operand isn’t an `NSEvaluatedObjectExpressionType` (i.e., `SELF`).